semgrep_metrics.atd

```diff
@@ -156,6 +156,33 @@ type pro_features = {
     ?diffDepth <ocaml mutable>: int option;
 }
 
+type analysis_type <ocaml attr="deriving show"> = [ 
+    | Intraprocedural 
+    | Interprocedural 
+    | Interfile
+]
+
+type secrets_config 
+     <ocaml attr="deriving show"> = {
+  permitted_origins: string;
+}
+
+type pro_config 
+     <ocaml attr="deriving show"> = {
+  analysis_type: analysis_type;
+  ?secrets_config: secrets_config option;
+  pro_langs: bool;
+}
```

Comment:
I'd like this to clearly be our central place where we report the configuration of products that we ran. e.g.
```
type products_config = {
    code: code_config option;
    secrets: secrets_config option;
    supply_chain: supply_chain_config option;
}
```

where `code_config` would be `analysis_type` and `supply_chain_config` can be `()` for now if there's no config. I want to make it really easy to answer the question "what products did x run with?"

The one thing is I realize that `analysis_type` affects both Code and Secrets. Frankly, I think it shouldn't, and we should have them configured separately. Because of that preference, for now, I would make analysis_type separately a part of secrets_config.

Reply:
> where code_config would be analysis_type and supply_chain_config can be () for now if there's no config. I want to make it really easy to answer the question "what products did x run with?"

Sure. I can add code and supply chain here. I didn't since there were not obvious configuration options to me which needed to be associated with code and from asking around it I was told there were not relevant options for supply chain to add here.

> The one thing is I realize that analysis_type affects both Code and Secrets. Frankly, I think it shouldn't, and we should have them configured separately.

Like I mentioned, I disagree. I think it just makes running rules more complicated if we allow mixing things like this. If we're already doing the naming resolution, e.g., we should just have everything benefit from that instead of restricting it and having to ensure it isn't used for only some rules.

Reply:
> Like I mentioned, I disagree. I think it just makes running rules more complicated if we allow mixing things like this. If we're already doing the naming resolution, e.g., we should just have everything benefit from that instead of restricting it and having to ensure it isn't used for only some rules.

Reflected on this more. I think what I really dislike right now is that it isn't clear that analysis_type affects both when you go to the toggle in the app. I'd be fine with having a `global_config` for options that affect all products